### ISSUE TITLE
EB 3.2, issues with bzip2 for GCCcore 6.3 - fine in EB 3.1.2 

### ISSUE THREAD

**Issue by jhein32** [hpcugent/easybuild-framework#2210]
Hi,

First a big thanks to everyone who contributed to EB 3.2.  I have just upgraded and am trying to install some of the new packages.  On our installation a simple build of bzip2 for GCCcore 6.3 fails quite late.  I get

```
-bash-4.2$ eb bzip2-1.0.6-GCCcore-6.3.0.eb
== temporary log file in case of crash /tmp/eb-I8DFVf/easybuild-5ULiML.log
== processing EasyBuild easyconfig /sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_easyconfigs-3.2.0-py2.7.egg/easybuild/easyconfigs/b/bzip2/bzip2-1.0.6-GCCcore-6.3.0.eb
== building and installing Compiler/GCCcore/6.3.0/bzip2/1.0.6...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== testing...
== installing...
== taking care of extensions...
== postprocessing...
== sanity checking...
ERROR: Traceback (most recent call last):
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_framework-3.2.0-py2.7.egg/easybuild/main.py", line 117, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_framework-3.2.0-py2.7.egg/easybuild/framework/easyblock.py", line 2531, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_framework-3.2.0-py2.7.egg/easybuild/framework/easyblock.py", line 2447, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_framework-3.2.0-py2.7.egg/easybuild/framework/easyblock.py", line 2325, in run_step
    step_method(self)()
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_easyblocks-3.2.0-py2.7.egg/easybuild/easyblocks/b/bzip2.py", line 110, in sanity_check_step
    super(EB_bzip2, self).sanity_check_step(custom_paths=custom_paths)
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_framework-3.2.0-py2.7.egg/easybuild/framework/easyblock.py", line 1844, in sanity_check_step
    self._sanity_check_step(*args, **kwargs)
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_framework-3.2.0-py2.7.egg/easybuild/framework/easyblock.py", line 2041, in _sanity_check_step
    fake_mod_data = self.load_fake_module(purge=True)
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_framework-3.2.0-py2.7.egg/easybuild/framework/easyblock.py", line 1211, in load_fake_module
    fake_mod_path = self.make_module_step(fake=True)
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_framework-3.2.0-py2.7.egg/easybuild/framework/easyblock.py", line 2149, in make_module_step
    txt += self.make_module_description()
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_framework-3.2.0-py2.7.egg/easybuild/framework/easyblock.py", line 1000, in make_module_description
    return self.module_generator.get_description()
  File "/sw/easybuild/software/Core/EasyBuild/3.2.0/lib/python2.7/site-packages/easybuild_framework-3.2.0-py2.7.egg/easybuild/tools/module_generator.py", line 697, in get_description
    'homepage': self.app.cfg['homepage'],
ValueError: unsupported format character 't' (0x74) at index 150
```

Downgrading to EB 3.1.2 I can build a bzip2 in GCCcore 6.3.0.  A diff shows no difference for the .eb files in EB 3.1.2 and 3.2.0

A diff shows the module_generator.py has been extensively reworked between EB 3.1.2 and 3.2.

I can build other software with EB 3.2, so it is not all broken.

Best wishes
   Joachim

**Comment by JensTimmerman** [comment]
This is a bug indeed, the '%' signs in the bzip2 description are not escaped before the description string is formatted.

```
description = """bzip2 is a freely available, patent free, high-quality data compressor. It typically
compresses files to within 10% to 15% <snip>"""
```

**Comment by boegel** [comment]
I noticed this issue over the weekend, fixed in #2209.

I'll issue a version 3.2.1 release later this week, just taking the opportunity to include some other things as well.
